app/controllers/catalog_controller.rb

```diff
@@ -41,6 +42,48 @@ def update_user_per_cat_page
     end
   end
 
+  def submit_form # rubocop:disable MethodLength, AbcSize
```

Comment:
This name is a bit too vague; let's make it a little more specific so that we know what it's for (updating the cart during confirmation... but not in so many words :stuck_out_tongue:).